x-pack/plugins/upgrade_assistant/__jest__/client_integration/cluster.test.ts

```diff
@@ -0,0 +1,278 @@
+
+      component.update();
+    });
+
+    test('renders deprecations', () => {
+      const { exists } = testBed;
+      expect(exists('clusterTabContent')).toBe(true);
+      expect(exists('deprecationsContainer')).toBe(true);
+    });
+
+    describe('fix ml snapshots button', () => {
+      let flyout: Element | null;
+
+      beforeEach(async () => {
+        const { component, actions, exists, find } = testBed;
+
+        expect(exists('deprecationsContainer')).toBe(true);
+
+        // Open all deprecations
+        actions.clickExpandAll();
+
+        // The data-test-subj is derived from the deprecation message
+        const accordionTestSubj = `depgroup_${upgradeStatusMockResponse.cluster[0].message
+          .split(' ')
+          .join('_')}`;
+
+        await act(async () => {
+          find(`${accordionTestSubj}.fixMlSnapshotsButton`).simulate('click');
+        });
+
+        component.update();
+
+        // We need to read the document "body" as the flyout is added there and not inside
+        // the component DOM tree.
+        flyout = document.body.querySelector('[data-test-subj="fixSnapshotsFlyout"]');
+
+        expect(flyout).not.toBe(null);
+        expect(flyout!.textContent).toContain('Upgrade or delete model snapshot');
+      });
+
+      test('upgrades snapshots', async () => {
+        const { component } = testBed;
+
+        const upgradeButton: HTMLButtonElement | null = flyout!.querySelector(
+          '[data-test-subj="upgradeSnapshotButton"]'
+        );
+
+        httpRequestsMockHelpers.setUpgradeMlSnapshotResponse({
+          acknowledged: true,
+        });
+
+        await act(async () => {
+          upgradeButton!.click();
+        });
+
+        component.update();
+
+        const request = server.requests[server.requests.length - 1];
+
+        expect(request.method).toBe('POST');
+        expect(request.url).toBe('/api/upgrade_assistant/ml_snapshots');
+        expect(request.status).toEqual(200);
+      });
+
+      test('deletes snapshots', async () => {
+        const { component } = testBed;
+
+        const deleteButton: HTMLButtonElement | null = flyout!.querySelector(
+          '[data-test-subj="deleteSnapshotButton"]'
+        );
+
+        httpRequestsMockHelpers.setDeleteMlSnapshotResponse({
+          acknowledged: true,
+        });
+
+        await act(async () => {
+          deleteButton!.click();
+        });
+
+        component.update();
+
+        const request = server.requests[server.requests.length - 1];
+        const mlDeprecation = upgradeStatusMockResponse.cluster[0];
+
+        expect(request.method).toBe('DELETE');
+        expect(request.url).toBe(
+          `/api/upgrade_assistant/ml_snapshots/${
+            (mlDeprecation.correctiveAction! as MlAction).jobId
+          }/${(mlDeprecation.correctiveAction! as MlAction).snapshotId}`
+        );
+        expect(request.status).toEqual(200);
+      });
+    });
+  });
+
+  describe('no deprecations', () => {
```

Comment:
This test coverage is currently repetitive of what is covered for the `Indices` tab. We're exploring having a single table for showing cluster/index deprecations with a filter option, in which case these tests would be refactored.